File: App 4/app/src/main/java/y/aula/app_5/MainActivity.java
package y.aula.app_5;

import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    Button saveNameButton, listButton;
    EditText varName;
    Intent otherActivity,intentItems;
    ArrayList<String> names = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        saveNameButton = (Button) findViewById(R.id.saveButton);
        listButton = (Button) findViewById(R.id.listButton);
        varName = (EditText) findViewById(R.id.textName);
        otherActivity = new Intent(MainActivity.this,ListActivity.class);

        saveNameButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                names.add(varName.getText().toString());
            }
        });

        listButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                otherActivity.putExtra("myArray",names);
                startActivity(otherActivity);
            }
        });
    }
}

File: App 1/app/src/main/java/y/app_1/MainActivity.kt
package y.app_1

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        buttonPage2.setOnClickListener {
            startActivity(Intent(this, Tela2::class.java))
        }
    }
}
